src/lib.rs

```diff
@@ -428,6 +428,22 @@ impl Build {
         self
     }
 
+    /// removes a flag from self.flags that was added by Build.flag()
+    ///
+    /// # example
+    /// ```no_run
+    /// cc::Build::new()
+    ///     .file("src/foo.c")
+    ///     .flag("unwanted_flag")
+    ///     .remove_flag("unwanted_flag")
+    ///     .compile("bar");
+    /// ```
+
+    pub fn remove_flag(&mut self, flag: &str) -> &mut Build {
+        self.flags.retain(|other_flag| &**other_flag == flag);
```

Comment:
I didn't noticed that you changed `!=` to `==`, `retain` only keeps element that returns `true` so it should be `!=`:

```suggestion
        self.flags.retain(|other_flag| &**other_flag != flag);
```